pymatviz/phonons.py

```diff
@@ -135,9 +134,15 @@ def _shaded_range(
         return fig
 
     shade_defaults = dict(layer="below", row="all", col="all")
-    y_lim = dict(zip(("y_min", "y_max"), fig.layout.yaxis.range, strict=True))
+    y_lim: dict[float | Literal["y_min", "y_max"], Any] = dict(
+        zip(("y_min", "y_max"), fig.layout.yaxis.range, strict=True),
+    )
 
+    # DEBUG: why (0, "y_min")
     shaded_ys = shaded_ys or {(0, "y_min"): dict(fillcolor="gray", opacity=0.07)}
```

Comment:
@janosh Is this intended or a typo (`shaded_ys = shaded_ys or {(0, "y_min"): dict(fillcolor="gray", opacity=0.07)}`)?

This doesn't conform with the type `shaded_ys: dict[tuple[YMin, YMax], dict[str, Any]] | bool | None`.

Should the key be `(0, "y_max")` (**I didn't look very closely**, just ask)?

Reply:
thanks for checking! the value is correct here, it's supposed to add a slight gray shading for negative phonon frequencies to visually highlight them (since they indicate dynamic instability of the corresponding structure). but the type is wrong. it should be `dict[tuple[YMin | YMax, YMin | YMax], dict[str, Any]]` since it doesn't matter to plotly whether the first or last value is larger, it's a valid range either way

Reply:
Thanks for the explanation! Would fix this!